code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_miner.dm

```diff
@@ -160,11 +160,9 @@ Difficulty: Medium
 		changeNext_move(CLICK_CD_RANGE)
 
 /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/proc/quick_attack_loop()
-	if(next_move <= world.time)
-		stoplag(1)
-		.() //retry
-		return
-	sleep((next_move - world.time) * 1.5)
+	while(next_move <= world.time) //this is done this way because next_move can change to be sooner while we sleep.
+		stoplag(1) 
+	sleep((next_move - world.time) * 1.5) //but don't ask me what the fuck this is about
```

Comment:
what the fuck is this about

Reply:
the chain is AttackingTarget(), which sets next_move, then this proc, which delays for what next_move was set to but 50% longer, then it tries to attack the target again, which repeats if successful.

Reply:
do_after is your friend

Reply:
this is not at all what do_after does, since the miner is moving around constantly.